### PR TITLE
Analytics: track annotation source and label filter toggles with `grid_filter_toggled`

### PR DESCRIPTION
**Pull request by LeonardoRosaa**
## What has changed and why?

Fires a `grid_filter_toggled` PostHog event when the user selects or deselects an annotation source or annotation label in the grid filter panel.

## How has it been tested?

Unit tests

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

## Summary by CodeRabbit

- **New Features**
  - Added analytics tracking for annotation source and label filter toggles, including selected/unselected action, value, active count, and collection context.
  - Centralized annotation source selection-change handling to ensure consistent tracking.
  - Updated label filter toggle wiring to pass the current collection ID.

- **Bug Fixes**
  - Prevents analytics events from firing when the toggled selection can’t be resolved to a known label/item.

- **Tests**
  - Expanded unit and interaction tests to verify correct analytics payloads and edge cases.

### CODE REVIEW

File: lightly_studio_view/src/lib/hooks/useAnnotationsFilter/useAnnotationsFilter.ts [hunk 1]
```diff
@@ -164,9 +167,22 @@ export function useAnnotationsFilter({
...
-            toggleSelectedAnnotationFilterId(labelId);
-        }
+        if (!labelId) return;
+
+        const currentIds = get(selectedAnnotationFilterIds);
+        const action = currentIds.has(labelId) ? 'unselected' : 'selected';
+
+        toggleSelectedAnnotationFilterId(labelId);
+
+        const active_count = get(selectedAnnotationFilterIds).size;
```
[the remark this answers is not in the document]
Comment (LeonardoRosaa) [reply lightly_studio_view/src/lib/hooks/useAnnotationsFilter/useAnnotationsFilter.ts]: Renamed

File: lightly_studio_view/src/lib/hooks/useAnnotationsFilter/useAnnotationsFilter.ts [hunk 2]
```diff
@@ -164,9 +167,22 @@ export function useAnnotationsFilter({
...
+
+        trackEvent('grid_filter_toggled', {
+            collection_id: collectionId,
```
[the remark this answers is not in the document]
Comment (LeonardoRosaa) [reply lightly_studio_view/src/lib/hooks/useAnnotationsFilter/useAnnotationsFilter.ts:180]: Good catch - changed the `toggleAnnotationFilterSelection` function so that it accepts `collectionId` as one of the parameters

**Review by IgorSusmelj — APPROVED** [review]
LGTM
